node/actors/direct_start_agent.go

```diff
@@ -32,17 +44,99 @@ func (a *DirectStartAgent) PostStop(ctx context.Context) error {
 }
 
 func (a *DirectStartAgent) Receive(ctx *goakt.ReceiveContext) {
```

Comment:
the receive function is a bit busy... I think it's cleaner to stay high level in the receive function and then call out to helpers in the `switch`, making it easier to skim through the list of messages being handled.